pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -280,6 +280,15 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		return reconcile.Result{}, errors.Wrap(err, "failed to create provisioner")
 	}
 
+	ready, err := prov.IsReady()
+	if !ready {
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err,
+				fmt.Sprintf("failed to check provisioner dependencies status"))
+		}
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
Agree, I noticed during the tests that the requeue time was pretty short, but wasn't sure about the amount. The `RequeueAfter` seems a good idea

Reply:
Initially I excluded the usage of provisioner.Result due the `Dirty` flag - which was not useful for the `IsReady()` method. Its adoption could make anyhow the interface more uniform, let's see how it will look like

Reply:
Done